stdlib/mexpr/ast.mc

```diff
@@ -396,6 +395,448 @@ lang MExprAst =
   TupleTypeAst + RecordTypeAst + DataTypeAst + ArithTypeAst + BoolTypeAst +
   AppTypeAst
 
+-----------------------
+-- BUILDER FUNCTIONS --
+-----------------------
```

Comment:
I can't remember all the previous discussions regarding the ast-builder, but I recall that it was a conscious decision to keep the `ast-builder.mc` functionality separated from the `ast.mc` functionality. The only reason I can think about at the moment is that including `ast.mc` should only include that AST and nothing else.

Reply:
Yes, I also vaguely remember such discussions. Let's have it again at the next meeting :)

Reply:
The merge is now reverted for this PR